hydrolib/core/io/structure/models.py

```diff
@@ -395,6 +409,23 @@ class Pump(Structure):
         "head",
         "reductionfactor",
     )
+    _orientation_validator = get_enum_validator("orientation", enum=Orientation)
+
+    _check_list_length = make_list_length_root_validator(
+        "startlevelsuctionside",
+        "stoplevelsuctionside",
+        "startleveldeliveryside",
+        "stopleveldeliveryside",
+        length_name="numstages",
+        list_required_with_length=True,
+    )
```

Comment:
Please correct me if I am wrong but I understand from the issue that if e.g. `numstages == 0` than all those fields are allowed to be `None` . But I don't see where that is handled.

Reply:
I've changed it into `elif list_required_with_length and length > 0:`